src/packages/app-framework/src/data.ts

```diff
@@ -250,3 +250,36 @@ export const getInstallationsImpl: GetInstallations = async ({ tableName }) => {
   });
   return result;
 };
+
+export type GetInstallationsByNodeId = ({
+  nodeId,
+  tableName,
+}: {
+  nodeId: string;
+  tableName: string;
+}) => Promise<InstallationRecord[]>;
+
+export const getInstallationsDataByNodeId: GetInstallationsByNodeId = async ({
+  nodeId,
+  tableName,
+}) => {
+  const getInstallations = new TableOperations({ TableName: tableName });
+  const result: InstallationRecord[] = [];
+  const itemList = await getInstallations.scan();
+  itemList.map((item) => {
+    const appId: number = item.AppId;
+    const installationId: number = item.InstallationId;
+    const itemNodeId: string = item.NodeId ?? '';
```

Comment:
According to the table definition, I think this is backwards. The App ID and Node ID are the primary index, so the Node ID is guaranteed to always be there but the Installation ID is [strictly speaking from a DDB data modeling perspective] not. They _should_ all three always be there, though.